ethsigner/core/src/main/java/tech/pegasys/ethsigner/core/requesthandler/sendtransaction/transaction/PrivacyIdentifier.java

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright 2019 ConsenSys AG.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+package tech.pegasys.ethsigner.core.requesthandler.sendtransaction.transaction;
+
+import java.nio.charset.StandardCharsets;
+import java.util.Arrays;
+import java.util.Base64;
+
+import org.web3j.utils.Numeric;
+
+public class PrivacyIdentifier {
+
+  public static final int IDENTIFIER_LENGTH = 32;
+
+  private final byte[] identifier;
+
+  public PrivacyIdentifier(byte[] identifier) {
```

Comment:
Can this constructor be made private so that consumers are forced to use the from methods?

Reply:
Done.